File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/FlingEvent.java
package sk.uxtweak.uxmobile.model.event;

import org.json.JSONArray;
import org.json.JSONException;

/**
 * Created by matej on 3.11.2017.
 */

public class FlingEvent extends TouchEvent {
    protected static final int INDEX_VELOCITY_X = 4;
    protected static final int INDEX_VELOCITY_Y = 5;

    private final double mVelocityX;
    private final double mVelocityY;

    public FlingEvent(long startTime, double x, double y, double velocityX, double velocityY) {
        super(startTime, x, y);
        mVelocityX = velocityX;
        mVelocityY = velocityY;
    }

    @Override
    protected String getType() {
        return TYPE_FLING;
    }

    @Override
    public JSONArray toJson() throws JSONException {
        return super.toJson()
                .put(INDEX_VELOCITY_X, mVelocityX)
                .put(INDEX_VELOCITY_Y, mVelocityY);
    }
}

File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/FlingEvent.kt
package sk.uxtweak.uxmobile.model.event

import org.json.JSONArray

class FlingEvent(
    startTime: Long,
    x: Float,
    y: Float,
    private val velocityX: Float,
    private val velocityY: Float
) : TouchEvent(startTime, x, y) {
    override fun getType() = TYPE_FLING

    override fun toJson(): JSONArray = super.toJson()
        .put(INDEX_VELOCITY_X, velocityX)
        .put(INDEX_VELOCITY_Y, velocityY)

    companion object {
        const val INDEX_VELOCITY_X = 4
        const val INDEX_VELOCITY_Y = 5
    }
}
